crml/nft/src/lib.rs

```diff
@@ -0,0 +1,651 @@
+			}
+			<Tokens<T>>::insert(&collection_id, token_id, token);
+			<NextTokenId<T>>::insert(&collection_id, next_token_id);
+			<TokenIssuance<T>>::mutate(&collection_id, |i| *i += One::one());
+			<TokenOwner<T>>::insert(&collection_id, token_id, owner.clone());
+			<CollectedTokens<T>>::append(&collection_id, owner.clone(), token_id);
+
+			Self::deposit_event(RawEvent::CreateToken(collection_id, token_id, owner));
+
+			Ok(())
+		}
+
+		/// Transfer ownership of an NFT
+		/// Caller must be the token owner
+		#[weight = T::WeightInfo::transfer()]
+		fn transfer(origin, collection_id: CollectionId, token_id: T::TokenId, new_owner: T::AccountId) {
+			let origin = ensure_signed(origin)?;
+
+			ensure!(CollectionSchema::contains_key(&collection_id), Error::<T>::NoCollection);
+			ensure!(<Tokens<T>>::contains_key(&collection_id, token_id), Error::<T>::NoToken);
+
+			let current_owner = Self::token_owner(&collection_id, token_id);
+			ensure!(current_owner == origin, Error::<T>::NoPermission);
+
+			ensure!(!<Listings<T>>::contains_key(&collection_id, token_id), Error::<T>::TokenListingProtection);
+
+			Self::transfer_ownership(&collection_id, token_id, &current_owner, &new_owner);
+			Self::deposit_event(RawEvent::Transfer(collection_id, token_id, new_owner));
+		}
+
+		/// Burn an NFT 🔥
+		/// Caller must be the token owner
+		#[weight = T::WeightInfo::burn()]
+		fn burn(origin, collection_id: CollectionId, token_id: T::TokenId) {
+			let origin = ensure_signed(origin)?;
+
+			ensure!(CollectionSchema::contains_key(&collection_id), Error::<T>::NoCollection);
+			ensure!(<Tokens<T>>::contains_key(&collection_id, token_id), Error::<T>::NoToken);
+
+			let current_owner = Self::token_owner(&collection_id, token_id);
+			ensure!(current_owner == origin, Error::<T>::NoPermission);
+
+			ensure!(!<Listings<T>>::contains_key(&collection_id, token_id), Error::<T>::TokenListingProtection);
+
+			// Update token ownership
+			<CollectedTokens<T>>::mutate(&collection_id, current_owner, |tokens| {
+				tokens.retain(|t| t != &token_id)
+			});
+			<TokenOwner<T>>::take(&collection_id, token_id);
+			<Tokens<T>>::take(&collection_id, token_id);
+
+			// Will not overflow, cannot exceed the amount issued qed.
+			let tokens_burnt = Self::tokens_burnt(&collection_id).checked_add(&One::one()).unwrap();
+			<TokensBurnt<T>>::insert(&collection_id, tokens_burnt);
+
+			Self::deposit_event(RawEvent::Burn(collection_id, token_id));
+		}
+
+		/// Sell an NFT to specific account at a fixed price
+		/// `buyer` optionally, the account to receive the NFT. If unspecified, then any account may purchase
+		/// `asset_id` fungible asset Id to receive as payment for the NFT
+		/// `fixed_price` ask price
+		/// Caller must be the token owner
+		#[weight = T::WeightInfo::direct_sale()]
+		fn direct_sale(origin, collection_id: CollectionId, token_id: T::TokenId, buyer: Option<T::AccountId>, payment_asset: AssetId, fixed_price: Balance) {
```

Comment:
I would suggest the duration to be an optional input parameter. Especially important for the fixed price sale.